compiler/damlc/daml-visual/src/DA/Daml/Visual.hs

```diff
@@ -67,6 +75,61 @@ data Graph = Graph
     , edges :: [(ChoiceDetails, ChoiceDetails)]
     } deriving (Show, Eq)
 
+data D3Link = D3Link
+    { source :: Int
+    , target :: Int
+    , weight :: Int
+    } deriving (Generic, Show)
+
+-- can add more information like signatories, observers
+data D3Node = D3Node
+    { fields :: T.Text
+    , tplName :: T.Text
+    , id :: Int
+    , chcName :: T.Text
+    } deriving (Generic, Show)
+
+data D3Graph = D3Graph
+    { d3links :: [D3Link]
+    , d3nodes :: [D3Node]
+    } deriving (Generic, Show)
+
+data WebPage = WebPage
+    { links :: T.Text
+    , dnodes :: T.Text
+    , d3Js :: String
+    , d3PlusJs :: String
+    } deriving (Generic, Show)
+
+d3LinksFromGraphEdges :: Graph -> [D3Link]
+d3LinksFromGraphEdges g = map edgeToD3Link (edges g)
+    where edgeToD3Link edge = D3Link (nodeId (fst edge)) (nodeId (snd edge)) 10
```

Comment:
The weight is not strictly necessary, but giving a default here saves me writing some JS. In realiity weight is used to determine the thickness of the arrow in the network svg